test/kafka-ingest-open-loop/mzcompose.py

```diff
@@ -0,0 +1,187 @@
+# upsert Kafka sources with 10 MM unique keys and 2kb values", and comparing the peak QPS
+# observed across different such workloads (along with the corresponding dataflow timing
+# info) can give insight into the relative overheads of different workloads and what
+# might be causing them.
+#
+# [1]: https://www.scylladb.com/2021/04/22/on-coordinated-omission/
+def workflow_kafka_ingest_open_loop(
+    c: Composition, parser: WorkflowArgumentParser
+) -> None:
+    parser.add_argument("--num-seconds", type=int, default=100)
+    parser.add_argument("--records-per-second", type=int, default=10000)
+    parser.add_argument("--num-keys", type=int, default=1000000000)
+    parser.add_argument("--value-bytes", type=int, default=500)
+    parser.add_argument("--timeout-secs", type=int, default=120)
+    parser.add_argument("--enable-persistence", action="store_true")
+    args = parser.parse_args()
+
+    c.start_and_wait_for_tcp(services=prerequisites)
+    mz_service = "materialized"
+    if args.enable_persistence:
+        mz_service = "materialized_persist"
+
+    c.up(mz_service)
+    c.wait_for_materialized(mz_service)
+
+    c.run(
+        "testdrive-svc",
+        "setup.td",
+    )
+
+    start = time.monotonic()
+    records_sent = 0
+    total_records_to_send = args.records_per_second * args.num_seconds
+    # Maximum observed delta between records sent by the benchmark and ingested by
+    # Materialize.
+    max_lag = 0
+    last_reported_time = 0.0
+
+    while True:
+        elapsed = time.monotonic() - start
+        records_ingested = query_materialize(c, mz_service)
+
+        lag = records_sent - records_ingested
+
+        if lag > max_lag:
+            max_lag = lag
+
+        # Report our findings back once per second.
+        if elapsed - last_reported_time > 1:
+            say(
+                f"after {elapsed:.3f}s sent {records_sent} records, and ingested {records_ingested}. max observed lag {max_lag} records, most recent lag {lag} records"
+            )
+            last_reported_time = elapsed
+
+        # Determine how many records we are scheduled to send, based on how long
+        # the benchmark has been running and the desired QPS.
+        records_scheduled = int(
+            min(elapsed, args.num_seconds) * args.records_per_second
+        )
+        records_to_send = records_scheduled - records_sent
+
+        if records_to_send > 0:
+            send_records(
+                c,
+                num_records=records_to_send,
+                num_keys=args.num_keys,
+                value_bytes=args.value_bytes,
+                timeout_secs=args.timeout_secs,
+            )
+            records_sent = records_scheduled
+
+        # Exit once we've sent all the records we need to send, and confirmed that
+        # Materialize has ingested them.
+        if (
+            records_sent == total_records_to_send
+            and records_ingested == total_records_to_send
+        ):
+            say(
```

Comment:
I think you just want `print` here. `say` is controlled by the `--mz-quiet` flag, but this output is important enough that you don't want to suppress it with the `--mz-quiet` flag. (Also, it's moot if you take this suggestion, but you want to import `say` from the `ui` module, not `mzcompose`.)